### ISSUE TITLE
`ens.utils.dict_copy` only operates on keyword arguments

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/web3.py#1423]
### What was wrong?

There is a `dict_copy` decorator used in the `ens` module.  It is intended to allow us to write methods like this:

```
@dict_copy
def my_method(self, some_params={}):
    ...
```

Normally using a mutable value like a dictionary as a default argument is problematic because the same mutable object is used across each call.  This decorator institutes a `copy` of this, but **only** if it is passed as a keyword argument.

There are at least some places where `dict_copy` is used but where the parameter that needs to be copied is not enforced to be a keyword argument, meaning that any calls that use it positionally will not benefit from the protection.

### How can it be fixed?

I'm inclined to suggest removing the decorator entirely and changing all places it was used to follow the standard pattern of:

```python
def my_method(self, some_params=None):
    if some_params is None:
        some_params = {}
    ...
```

**Comment by Patil2099** [comment]
Can I work on it? @pipermerriam?

**Comment by pipermerriam** [comment]
@Patil2099 :+1:

**Comment by shashankks0987** [comment]
I'd like to take this if it is still open!

**Comment by dbfreem** [comment]
So I was thinking about taking this one on but then started thinking. Wouldn't this be a breaking change as consumers of the library could be relying on the @dict_copy to not mutate the dictionary that they are passing in to things like main.setup_address. I can make the change but just wanted to make sure I made it on the right branch as it might not be appropriate to make the change on master.

**Comment by pacrob** [comment]
Fixed in PR #1998
